src/SourceBuild/content/eng/tools/PrBaselinePublisher/Publisher.cs

```diff
@@ -0,0 +1,374 @@
+            Content = content,
+            Encoding = EncodingType.Utf8
+        };
+        return await _client.Git.Blob.Create(_repoOwner, _repoName, blob);
+    }
+
+    private string ParseUpdatedFileName(string updatedFile)
+    {
+        return updatedFile.Split("Updated")[1];
+    }
+
+    private async Task<TreeResponse> CreateTreeFromItems(List<NewTreeItem> items, string path = "")
+    {
+        var newTreeItems = new List<NewTreeItem>();
+
+        var groups = items.GroupBy(item => Path.GetDirectoryName(item.Path));
+        foreach (var group in groups)
+        {
+            if (string.IsNullOrEmpty(group.Key) || group.Key == path)
+            {
+                // These items are in the current directory, so add them to the new tree items
+                foreach (var item in group)
+                {
+                    if(item.Type != TreeType.Tree)
+                    {
+                        newTreeItems.Add(new NewTreeItem
+                        {
+                            Path = path == string.Empty ? item.Path : Path.GetRelativePath(path, item.Path),
+                            Mode = item.Mode,
+                            Type = item.Type,
+                            Sha = item.Sha
+                        });
+                    }
+                }
+            }
+            else
+            {
+                // These items are in a subdirectory, so recursively create a tree for them
+                var subtreeResponse = await CreateTreeFromItems(group.ToList(), group.Key);
+                newTreeItems.Add(new NewTreeItem
+                {
+                    Path = group.Key,
+                    Mode = "040000",
+                    Type = TreeType.Tree,
+                    Sha = subtreeResponse.Sha
+                });
+            }
+        }
+
+        var newTree = new NewTree();
+        foreach (var item in newTreeItems)
+        {
+            newTree.Tree.Add(item);
+        }
+        return await _client.Git.Tree.Create(_repoOwner, _repoName, newTree);
+    }
+
+    private async Task<TreeResponse> CreateParentTree(TreeResponse testResultsTreeResponse, TreeResponse originalTreeResponse, string originalTestResultsPath)
+    {
+        // Create a new tree for the parent directory
+        // excluding anything in the updated test results tree
+        NewTree parentTree = new NewTree();
+        foreach (var file in originalTreeResponse.Tree)
+        {
+            if (!file.Path.Contains(originalTestResultsPath))
+            {
+                parentTree.Tree.Add(new NewTreeItem
+                {
+                    Path = file.Path,
+                    Mode = file.Mode,
+                    Type = file.Type.Value,
+                    Sha = file.Sha
+                });
+            }
+        }
+
+        //  Connect the updated test results tree
+        parentTree.Tree.Add(new NewTreeItem
+        {
+            Path = originalTestResultsPath,
+            Mode = "040000",
```

Comment:
Define as a constant with a descriptive name or at least include a comment that indicates what this value represents.